rfc/drafts/mediachain-rfc-3.md

```diff
@@ -0,0 +1,448 @@
+- **Search** -  End-user submits a media file or media ID, a list of most semantically relevant media IDs is returned.
+- **Dedupe** - Determine which media in the index are duplicates / non-duplicates. Submit a subset of these results to the `Transactors`. This process can be initiated periodically to reprocess all media in the search index - for example after each time the dedupe models are considerably re-trained. Or, it can act only on new media from the `Journal`, received via the `Client`.
+- **Model Training & Hyper-Parameter Tuning Lifecycle** - Background training and tuning process that updates the information retrieval / dedupe models and tunes hyper-parameters. After each re-training, models are regression tested, put into production where live-user regression metrics are again measured. If major regressions are detected, models and hyper-parameters are reverted to an earlier version.
+
+
+## Ingestion Subsystem
+
+The `Ingestion` subsystem accepts media works from the `Transactors` via the `Clients`, performs basic attack
+mitigation and media prioritization, and then processes the media objects into image feature descriptors. These
+feature descriptors, along with the media IDs are then inserted into the `KNN Index`, which will be queried by
+the `Search` and `Dedupe` systems.
+
+User interfaces to the `Ingestion`  subsystem will in coordination with the `Dedupe` subsystem, to alert users if they are
+about to create duplicate entries for the same work in the MediaChain blockchain, or allow users to provide a negative training
+signal to the deduplication subsystem in the case where media works have been incorrectly flagged as duplicates.
+
+In the case when an exact match of media content, which was not caught by the MediaChain core - for example images files with the same
+image content but differing EXIF headers, the `Ingestion` subsystem may, at insertion time, attempt to force these media works to resolve
+to the same blockchain ID.
+
+Future `Ingestion` subsystem enhancements include receiving indexing data directly from users who are inserting media works, in
+order to minimize indexing delays, and adding more sophisticated APIs for user feedback and querying.
+
+
+## Search Subsystem
+
+The `Search` system is exposed to end-users via a REST/JSON API. The address of the REST API will be communicated to end-users via the
+Mediachain Core. This API allows users to input a query or media ID, and receive a ranked list of most semantically-relevant media IDs.
+
+
+ Method | REST Call     | Description                                       | Returns
+ -------|---------------|---------------------------------------------------|---------------------------------------------------
+ POST   | `/search`     | Search by base64-encoded image or image ID.       | List of image IDs, possibly with relevancy scores.
+ POST   | `/distance`   | Calculate metric-space distance between 2 images. | Numeric distance between 0.0 and 1.0.
```

Comment:
Yes, that line got quite jumbled... In fact, let's rather have a similarity function there instead of a distance, and we'll change the description to "Calculate similarity score for a pair of images".